apps/api/src/audio/audio.service.ts

```diff
@@ -0,0 +1,82 @@
+import { Injectable, Logger } from "@nestjs/common";
+
+@Injectable()
+export class AudioService {
+  private readonly logger = new Logger(AudioService.name);
+
+  async transcribeAudio(
+    audioBuffer: Buffer,
+    format: string = "mp3",
+  ): Promise<string> {
+    if (audioBuffer.length === 0) {
+      throw new Error("Invalid audio data - buffer is empty");
+    }
+    try {
+      this.logger.log(
+        `Transcribing audio: format=${format}, bufferSize=${audioBuffer.length} bytes`,
+      );
+
+      const transcription = await this.transcribeWithOpenai(
+        audioBuffer,
+        format,
+      );
+
+      return this.processTranscriptionResult(transcription);
+    } catch (error: unknown) {
+      const errorMessage =
+        error instanceof Error ? error.message : "Unknown error";
+      this.logger.error(`Audio transcription failed: ${errorMessage}`);
+      throw new Error(`Failed to transcribe audio: ${errorMessage}`);
+    }
+  }
+
+  private async transcribeWithOpenai(
+    audioBuffer: Buffer,
+    format: string,
+    model: string = "gpt-4o-mini-transcribe",
+  ): Promise<string> {
+    const formData = new FormData();
+    const mimeType = format === "mp3" ? "audio/mpeg" : `audio/${format}`;
```

Comment:
wait, so in the other file you translated mime type -> a generic type like "wav" or "mp3" and then here you're translating it back to a mime type? Why not just use mime types all the way through... and then if there are wierd one-offs, handle it here